### DISCUSSION TITLE
[Feature Request] - Token Visibility

### DISCUSSION THREAD

**Discussion by ollieanwyll** [danny-avila/LibreChat#3861] [Feature Requests & Suggestions]
Hey Danny and team,

I was wondering if you could add a feature to show the number of tokens accumulated in a conversation. It would be nice to have visibility on this. Additionally, it would be great to have the option to see the token count in the text editor before sending a message. This feature could potentially be in the form of a page in the settings, where you can paste text into a larger window and get the token count. It would be similar to:

https://platform.openai.com/tokenizer
https://articlefiesta.com/seo-tools/token-counter

Thanks!

**Comment by anose001** [comment]
Came to the discussion forum to voice the above. Maybe even have a toggle to convert to $ cost based on Token cost that the user can set in some settings pane? I understand that the Token cost feature may be undesireable because token costs just change so quickly over time, but a token counter on a per message/respond basis would be great.

I am still unsure these days whether only the latest message counts towards the token count or the whole message history. Edit: This comment [here](https://community.openai.com/t/conversation-context-and-quadratic-billing/126421#:~:text=I%20think%20you,it%20works%2C%20yes.) suggest that all the histroy of the conversation counts towards the token count. This means that the cost of the next message send in a conversation increases non-linearly with each additional message. This can spiral out of hand for really long messages.
